ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -177,28 +180,24 @@ public struct SignalProducer<Value, Error: ErrorType> {
 		}
 
 		let bufferingObserver: Signal<Value, Error>.Observer = Observer { event in
-			// Send serially with respect to other senders, and never while
-			// another thread is in the process of replaying.
-			dispatch_sync(queue) {
-				let originalState = state.modify { state in
-					var mutableState = state
-
-					if let value = event.value {
-						mutableState.addValue(value, upToCapacity: capacity)
-					} else {
-						// Disconnect all observers and prevent future
-						// attachments.
-						mutableState.terminationEvent = event
-						mutableState.observers = nil
-					}
-
-					return mutableState
+			let originalState = state.modify { state in
+				var mutableState = state
+				
+				if let value = event.value {
+					mutableState.addValue(value, upToCapacity: capacity)
+				} else {
+					// Disconnect all observers and prevent future
+					// attachments.
+					mutableState.terminationEvent = event
+					mutableState.observers = nil
 				}
-
-				if let observers = originalState.observers {
-					for observer in observers {
-						observer.action(event)
-					}
+				
+				return mutableState
+			}
+			
+			if let observers = originalState.observers {
```

Comment:
This could be written as `originalState.observers?.forEach { $0.action(event) }`